opencog/moses/main/populate_atomspace.cc

```diff
@@ -12,9 +12,15 @@ void populate(AtomSpace *as, const ITable &itable) {
 	int n_columns = itable.get_types().size();
 	for (int i = 0; i < n_columns; i++) {
 		id::type_node col_type = itable.get_types().at(i);
-		const vertex_seq &col = itable.get_column_data(i);
-		const Handle &feature = createNode(SCHEMA_NODE,itable.get_labels().at(i));
-		const ValuePtr &vtr = vertex_seq_to_value(col, col_type);
+		vertex_seq col = itable.get_column_data(i);
+		Handle feature;
+		if (col_type == id::boolean_type){
+		feature = createNode(PREDICATE_NODE,itable.get_labels().at(i));
```

Comment:
Need to fix indentation.

Alternatively you could use a ternary conditional instead of `if`, as follows
```suggestion
		feature = createNode(col_type == id::boolean_type ? PREDICATE_NODE : SCHEMA_NODE,
		                     itable.get_labels().at(i));
```

Reply:
Are enum_types and other non boolean_types considered SCHEMA NODE?

Reply:
Good point. I suppose they will be `SCHEMA_NODE`. Indeed, if they were not this would lead to nested ternary conditionals which is not the most elegant construct, but I think most like it's only gonna be `SCHEMA_NODE`.

Sometimes I wonder if the boolean case shouldn't be also handled with `SCHEMA_NODE` but that's another topic.

Reply:
Yeah that's how i thought of it in the first place. which was why i refrained from using ternary conditionals since i was thinking to expand the if else block for other types. But if that is resolved we can go with the ternary like you suggested